### PR TITLE
FYLE-2891u2w-suggest

### PR DESCRIPTION
**Pull request by Kulsekar4**

### CODE REVIEW

**Review by gokulk16 — APPROVED** [review]

File: src/admin/paths/admin@statement_mappings@suggest.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            type: object
+            properties:
+              count:
+                $ref: '../../components/schemas/count.yaml'
+              offset:
+                $ref: '../../components/schemas/offset.yaml'
+              data:
+                type: object
+                items:
+                  $ref: '../../components/schemas/statement_mappings.yaml#/suggest_out'
```
Comment (modernwarfareuplink) [inline src/admin/paths/admin@statement_mappings@suggest.yaml:42]: pls to add `400` response also. For more info, pls check the GET vendors api

File: src/components/schemas/statement_mappings.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+suggest_out:
+  type: object
+  additionalProperties: False
+  properties:
+    id:
```
Comment (modernwarfareuplink) [inline src/components/schemas/statement_mappings.yaml:5]: `created_at` and `updated_at` are missing

File: src/components/schemas/statement_mappings.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    bank_name:
+      type: string
+      description: |
+        Specifies the bank name of the statement.
+      example: American Express
+    is_debit_positive:
+      type: boolean
+      description: |
+        Specifies the representation of the the amount in the statement, if true then the amount represented by 
+        negative sign are credits, and if false then the amount represented by negative sign are debits.
```
Comment (modernwarfareuplink) [inline src/components/schemas/statement_mappings.yaml:18]: p1 | Pls to inverse the statement, as the column name and the description is inverse and can cause strain on reader.
positive sign are debits, and if false then the amount represented by negative sign are debits.

File: src/components/schemas/statement_mappings.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
+suggest_out:
+  type: object
+  additionalProperties: False
+  properties:
+    id:
+      allOf:
+        - $ref: './fields.yaml#/id_string'
```
Comment (modernwarfareuplink) [inline src/components/schemas/statement_mappings.yaml:7]: `org_id` column is also missing

File: src/components/schemas/statement_mappings.yaml (hunk 2 above)
Comment (Kulsekar4) [reply src/components/schemas/statement_mappings.yaml:5]: Not needed for this api sir, as this is a suggestion api.

File: src/components/schemas/statement_mappings.yaml (hunk 4 above)
Comment (Kulsekar4) [reply src/components/schemas/statement_mappings.yaml:7]: Not needed for this api sir, as this is a suggestion api.

File: src/components/schemas/statement_mappings.yaml [hunk 5]
```diff
@@ -0,0 +1,87 @@
+suggest_out:
```
Comment (aditya999123) [inline src/components/schemas/statement_mappings.yaml:1]: statement_mappings_suggest_out

File: src/components/schemas/statement_mappings.yaml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      type: string
+      description: |
+        Specifies the header name of the description column in the statement.
+      nullable: true
+      example: description
+    is_auto_created:
+      type: boolean
+      description: |
+        Helps to identify if a statement mapping got auto created or not
+      example: true
```
Comment (aditya999123) [inline src/components/schemas/statement_mappings.yaml:73]: false

**Review by aditya999123 — APPROVED** [review]

**Pull request closed without merging**